processing/src/main/java/io/druid/query/groupby/epinephelinae/column/DoubleGroupByColumnSelectorStrategy.java

```diff
@@ -46,7 +45,19 @@ public void processValueFromGroupingKey(
   @Override
   public void initColumnValues(ColumnValueSelector selector, int columnIndex, Object[] values)
   {
-   values[columnIndex] = ((DoubleColumnSelector) selector).get();
+   values[columnIndex] = selector.getDouble();
+  }
+
+  @Override
+  public Object getOnlyValue(ColumnValueSelector selector)
+  {
+    return selector.getDouble();
+  }
+
+  @Override
+  public void writeToKeyBuffer(int keyBufferPosition, Object obj, ByteBuffer keyBuffer)
+  {
+    keyBuffer.putDouble(keyBufferPosition, (Double) obj);
   }
 
```

Comment:
@xanec, I added them in https://github.com/druid-io/druid/pull/4576, and guess it's possible to combine them. Are you interested in making a patch to fix it?